### PR TITLE
Make 01086_window_view_cleanup more stable

### PR DESCRIPTION
**Pull request by ilejn**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Makes 01086_window_view_cleanup more stable/less flaky.

### CODE REVIEW

**Comment by ilejn** [comment]
Appeared in https://github.com/ClickHouse/ClickHouse/pull/72123 

I failed to properly reproduce the issue, while as far as I understand

- the test is based on an assumption that cleaner (wake ups every second) would be fired during 2 seconds interval. 
- this is not always true in busy environments.
- if cleanup did not happen before last insert, StorageWindowView::getCleanupBound exits upon `max_fired_watermark == 0` condition, therefore clean up does not do anything, therefore loop
```
while true; do
	$CLICKHOUSE_CLIENT "${opts[@]}" --query="SELECT count(*) FROM ${DATABASE_ORDINARY}.\`.inner.wv\`" | grep -q "3" && break || sleep .5 ||:
done
```
is infinite (there are 6 rows in `.inner.wv`), therefore the test timeouts.

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit c60e7faba6b3e965ea470eb7049750c4400c3862 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/72232/c60e7faba6b3e965ea470eb7049750c4400c3862/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/fdd940b61800535de4a4252f2383b72df645e9f3/unit_tests__release_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by ilejn** [comment]
Hello @nikitamikhaylov , should I do anything further?

File: tests/queries/0_stateless/01086_window_view_cleanup.sh [hunk 1]
```diff
@@ -35,6 +35,7 @@ while true; do
 	$CLICKHOUSE_CLIENT "${opts[@]}" --query="SELECT count(*) FROM ${DATABASE_ORDINARY}.\`.inner.wv\`" | grep -q "5" && break || sleep .5 ||:
 done
 
+$CLICKHOUSE_CLIENT "${opts[@]}" --query="SELECT sleep(2);"
```
Comment (nikitamikhaylov) [inline tests/queries/0_stateless/01086_window_view_cleanup.sh:38]: FYI we don't appreciate tests that rely heavily on sleeps.

**Review by nikitamikhaylov — APPROVED** [review]

**Pull request merged**